tests/test_registry.py

```diff
@@ -37,13 +46,21 @@ def test_register_two_components(self):
             },
         )
 
-    def test_prevent_registering_twice(self):
+    def test_prevent_registering_different_components_with_the_same_name(self):
         self.registry.register(name="testcomponent", component=MockComponent)
         with self.assertRaises(component.AlreadyRegistered):
             self.registry.register(
-                name="testcomponent", component=MockComponent
+                name="testcomponent", component=MockComponent2
             )
 
+    def test_allow_duplicated_registration_of_the_same_component(self):
+        self.registry.register(
+            name="testcomponent", component=MockComponentView
+        )
+        self.registry.register(
+            name="testcomponent", component=MockComponentView
+        )
+
```

Comment:
I think all tests should have an assertion somewhere. For this test I suggest this pattern:

```python
    try:
        myFunc()
    except ExceptionType:
        self.fail("myFunc() raised ExceptionType unexpectedly!")
```